homeassistant/components/sensor/pollen.py

```diff
@@ -0,0 +1,349 @@
+ATTR_ALLERGEN_NAME = 'primary_allergen_name'
+ATTR_ALLERGEN_TYPE = 'primary_allergen_type'
+ATTR_CITY = 'city'
+ATTR_OUTLOOK = 'outlook'
+ATTR_RATING = 'rating'
+ATTR_SEASON = 'season'
+ATTR_TREND = 'trend'
+ATTR_ZIP_CODE = 'zip_code'
+
+CONF_ZIP_CODE = 'zip_code'
+
+DEFAULT_ATTRIBUTION = 'Data provided by IQVIA™'
+
+MIN_TIME_UPDATE_AVERAGES = timedelta(hours=12)
+MIN_TIME_UPDATE_INDICES = timedelta(minutes=10)
+
+CONDITIONS = {
+    'allergy_average_forecasted': (
+        'Allergy Index: Forecasted Average',
+        'AllergyAverageSensor',
+        'allergy_average_data',
+        {'data_attr': 'extended_data'},
+        'mdi:flower'
+    ),
+    'allergy_average_historical': (
+        'Allergy Index: Historical Average',
+        'AllergyAverageSensor',
+        'allergy_average_data',
+        {'data_attr': 'historic_data'},
+        'mdi:flower'
+    ),
+    'allergy_index_today': (
+        'Allergy Index: Today',
+        'AllergyIndexSensor',
+        'allergy_index_data',
+        {'key': 'Today'},
+        'mdi:flower'
+    ),
+    'allergy_index_tomorrow': (
+        'Allergy Index: Tomorrow',
+        'AllergyIndexSensor',
+        'allergy_index_data',
+        {'key': 'Tomorrow'},
+        'mdi:flower'
+    ),
+    'allergy_index_yesterday': (
+        'Allergy Index: Yesterday',
+        'AllergyIndexSensor',
+        'allergy_index_data',
+        {'key': 'Yesterday'},
+        'mdi:flower'
+    ),
+    'disease_average_forecasted': (
+        'Cold & Flu: Forecasted Average',
+        'AllergyAverageSensor',
+        'disease_average_data',
+        {'data_attr': 'extended_data'},
+        'mdi:snowflake'
+    )
+}
+
+RATING_MAPPING = [{
+    'label': 'Low',
+    'minimum': 0.0,
+    'maximum': 2.4
+}, {
+    'label': 'Low/Medium',
+    'minimum': 2.5,
+    'maximum': 4.8
+}, {
+    'label': 'Medium',
+    'minimum': 4.9,
+    'maximum': 7.2
+}, {
+    'label': 'Medium/High',
+    'minimum': 7.3,
+    'maximum': 9.6
+}, {
+    'label': 'High',
+    'minimum': 9.7,
+    'maximum': 12
+}]
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ZIP_CODE): cv.positive_int,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(CONDITIONS)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from pypollencom import Client
+
+    _LOGGER.debug('Configuration data: %s', config)
+
+    client = Client(config[CONF_ZIP_CODE])
+    datas = {
+        'allergy_average_data': AllergyAveragesData(client),
+        'allergy_index_data': AllergyIndexData(client),
+        'disease_average_data': DiseaseData(client)
+    }
+
+    for data in datas.values():
+        data.update()
+
+    sensors = []
+    for condition in config.get(CONF_MONITORED_CONDITIONS, []):
+        name, sensor_class, data_key, params, icon = CONDITIONS[condition]
+        sensors.append(globals()[sensor_class](
```

Comment:
I don't think you need to use `globals`. Is there a specific reason you want that?

Reply:
This so that I can dynamically instantiate objects of a certain type from the `CONDITIONS` dict. Open to better suggestions!

Reply:
Can't you just call `sensor_class` and instantiate? I think that should work.

Reply:
Sorry I wasn't clear. If you replace the string representing the class name in `CONDITIONS` with the actual class name you could call it directly from here. You might have to move the conditions dict below the class defines though, so that might be uglier. You decide.

Reply:
Wow! Didn't know you could do that. Thank you!

Reply:
Ah, I see. Putting `CONDITIONS` at the bottom of the file feels confusing to me (given all other constants are at the top). Given that, are you okay with me keeping as-is?

Reply:
Yeah, I'm ok with it.